src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -134,6 +138,8 @@ final class HashMap[K, +V] private[immutable] (private[immutable] val rootNode:
 
   override def foreach[U](f: ((K, V)) => U): Unit = rootNode.foreach(f)
 
+  private[collection] def foreachWithHash(f: (K, V, Int) => Unit): Unit = rootNode.foreachWithHash(f)
```

Comment:
Where do you think that logic should go? I was trying to at least somewhat minimize the amount of implementation details of `mutable.Hash*` exposed to `immutable.Hash*`

Reply:
What I mean is mark `foreachWithHash` as `@inline` and make sure the necessary members are `private[collection]`, not `private` or `private[this]`. The optimizer should even warn if it cannot inline the call because of access restrictions so it should be easy to find out what needs to be changed.

Reply:
Done! marked as ``` @`inline` ```